sql/src/main/java/io/crate/analyze/expressions/ExpressionToColumnIdentVisitor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.expressions;
+
+import io.crate.metadata.ColumnIdent;
+import io.crate.sql.tree.AstVisitor;
+import io.crate.sql.tree.Node;
+import io.crate.sql.tree.QualifiedNameReference;
+import io.crate.sql.tree.StringLiteral;
+import io.crate.sql.tree.SubscriptExpression;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class ExpressionToColumnIdentVisitor extends AstVisitor<Void, List<String>> {
+
+    private final static ExpressionToColumnIdentVisitor INSTANCE = new ExpressionToColumnIdentVisitor();
+
+    private ExpressionToColumnIdentVisitor() {
+    }
+
+    public static ColumnIdent convert(Node node) {
+        ArrayList<String> context = new ArrayList<>();
+        INSTANCE.process(node, context);
+        return ColumnIdent.fromList(context);
```

Comment:
probably it would make sense to use ColumnIdent as return value and pass the path of the columnIdent as context. this would not require all this list creations